library/direct_message_check.inc

```diff
@@ -517,22 +521,65 @@ function phimail_service_userID($name = 'phimail-service')
     return ($user);
 }
 
+/**
+ * Given an IsAcceptedFileFilterEvent from our isWhitelist function we check to make sure that we allow
+ * the mime type of the Direct message attachment that was sent by the server.  We check to make sure that the mime
+ * type is only the mime type of the specific document sent via EMRDirect
+ * @param IsAcceptedFileFilterEvent $event The event with the mimetype to check if we allow it or not
+ * @return IsAcceptedFileFilterEvent
+ */
+function phimail_allow_document_mimetype(IsAcceptedFileFilterEvent $event)
+{
+    global $phimail_direct_message_check_allowed_mimetype;
+    $isAllowedFile = $event->isAllowedFile();
+    if (!$isAllowedFile) {
+        if ($phimail_direct_message_check_allowed_mimetype == $event->getMimeType()) {
+            $event->setAllowedFile(true);
+        }
+    }
+    return $event;
+}
+
 
 /**
  * Registers an attachment or non-text message file using the existing Document structure
  * @return Array(doc_id,URL) of the file as stored in documents table, false = failure
  */
 function phimail_store($name, $mime_type, $fn)
 {
+    global $phimail_direct_message_check_allowed_mimetype;
 
+    $allowMimeTypeFunction = 'phimail_allow_document_mimetype';
+    // we bypass the whitelisting JUST for phimail documents
+    if (isset($GLOBALS['kernel'])) {
+        $GLOBALS['kernel']->getEventDispatcher()
+            ->addListener(IsAcceptedFileFilterEvent::EVENT_FILTER_IS_ACCEPTED_FILE, $allowMimeTypeFunction);
+    }
     // Collect phimail user id
     $user = phimail_service_userID();
-
-    // Import the document
-    $return = addNewDocument($name, $mime_type, $fn, 0, filesize($fn), $user, 'direct');
-
-    // Remove the temporary file
-    @unlink($fn);
+    try {
+        // Import the document
+        $phimail_direct_message_check_allowed_mimetype = $mime_type;
+        $filesize = filesize($fn);
+        $return = addNewDocument($name, $mime_type, $fn, 0, $filesize, $user, 'direct');
+        if (is_array($return)) {
+            $return['filesize'] = $filesize;
+        }
+    } catch (\Exception $exception) {
+        (new \OpenEMR\Common\Logging\SystemLogger())->errorLogCaller($exception->getMessage(), ['name' => $name, 'mime_type' => $mime_type, 'fn' => $fn]);
```

Comment:
place in use block above so can use shortened form

Reply:
Fixed